personal_finances/user/token_validation.py

```diff
@@ -0,0 +1,91 @@
+import json
+import logging
+from typing import Any
+import boto3
+import jwt
+import os
+
+LOGGER = logging.getLogger(__name__)
+
+
+class InvalidLambdaEventInput(Exception):
+    pass
+
+
+class AuthHeaderNotFound(Exception):
+    pass
+
+
+class EmptyUserID(Exception):
+    pass
+
+
+class AuthorizationHeaderEmptyContent(Exception):
+    pass
+
+
+def _get_token_from_event(event: dict) -> Any:
+    if "authorizationToken" not in event:
+        raise (AuthHeaderNotFound)
+
+    token = event.get("authorizationToken", None)
+    if token is None or token == "":
+        raise (AuthorizationHeaderEmptyContent)
+
+    return token
+
+
+def _get_jwt_secret() -> str:
+    secret_id = os.environ["INFINEAT_JWT_SECRET_NAME"]
+
+    jwt_session = boto3.client("secretsmanager")
+    get_secret_value_response = jwt_session.get_secret_value(SecretId=secret_id)
+
+    return str(get_secret_value_response["SecretString"])
+
+
+def _generatePolicy(principalId: str, effect: str, resource: str) -> str:
+    authResponse: dict = {}
+    authResponse["principalId"] = principalId
+    if effect and resource:
+        policyDocument: dict = {}
+        policyDocument["Statement"] = []
+        statementOne = {}
+        statementOne["Action"] = "execute-api:Invoke"
+        statementOne["Effect"] = effect
+        statementOne["Resource"] = resource
+        policyDocument["Statement"] = [statementOne]
+        authResponse["policyDocument"] = policyDocument
+    authResponse_JSON = json.dumps(authResponse)
+    return authResponse_JSON
```

Comment:
We can do simpler here, effect and resource are not optional according to function signature.
```
    return json.dumps({
        "policyDocument": {
            "principalId": principalId,
            "Statement": [{
                "Action": "execute-api:Invoke",
                "Effect": effect,
                "Resource": resrouce,
            }]
        }
    })
```